src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -785,20 +1176,26 @@ private void doMqttConnack(
                 .properties(properties)
                 .build();
 
-            doData(connack);
+            connected = true;
```

Comment:
This should be set in `onDecodeConnect` instead since we are not really `connected` when sending error on failed initial connect.